R/teal_data-class.R

```diff
@@ -74,6 +76,7 @@ new_teal_data <- function(data,
   if (length(code)) {
     code <- paste(code, collapse = "\n")
   }
+  valid <- !(length(code) || !identical(list(), data))
```

Comment:
```suggestion
  valid <- (length(code) == 0L && length(data) == 0L)
```

Reply:
@chlebowa what about 

```r
valid <- !(length(code) + length(data))
```

Reply:
Valid but a bit ugly IMO.

Reply:
We have `length(code)` used in `if` statement just a line above
https://github.com/insightsengineering/teal.data/blob/38d2b8de0e95712fa60d13e6f347a7d48c3d9b64/R/teal_data-class.R#L76-L78

In this case it's just `length(code) + length(data)` but we need to negate it.

`!(length(code) + length(data))` is not that bad, but I'm having second thoughts. It takes me 5 seconds to realize what's going on, where `(length(code) == 0L && length(data) == 0L)` is understandable in a flash : ) will update with your suggestion